### PR TITLE
Deprecate defaultProject

### PR DESCRIPTION
**Pull request by filipesilva**
More context in https://github.com/angular/angular-cli/issues/11111#issuecomment-425176217

Fix https://github.com/angular/angular-cli/issues/11111

### CODE REVIEW

File: packages/angular_devkit/core/src/workspace/workspace.ts [hunk 1]
```diff
@@ -140,6 +140,7 @@ export class Workspace {
     return workspaceProjectClone;
   }
 
+  // `defaultProject` is deprecated and should be removed in CLI v8.
   getDefaultProjectName(): string | null {
```
Comment (alan-agius4) [inline packages/angular_devkit/core/src/workspace/workspace.ts]: Can you use `@deprecated`? As when using it tslint is able to warn users. Same thing in the IDE

https://palantir.github.io//tslint/rules/deprecation/

File: packages/angular_devkit/core/src/workspace/workspace-schema.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@ export interface WorkspaceSchema {
    */
   newProjectRoot?: string;
   /**
-   * The default project.
+   * (DEPRECATED) The default project.
```
Comment (alan-agius4) [inline packages/angular_devkit/core/src/workspace/workspace-schema.ts]: Same

File: docs/documentation/angular-workspace.md [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
 - **version** (`integer`): File format version. This is currently `"1"`.
 
-- **newProjectRoot** (`string`): Path where new projects will be created.
+- **newProjectRoot** (`string`): (**DEPRECATED**) Path where new projects will be created.
```
Comment (alexeagle) [inline docs/documentation/angular-workspace.md]: this looks wrong, shouldn't it be on the following option?

Comment (filipesilva) [reply docs/documentation/angular-workspace.md]: Jeez, that's what I get for amending commits in a rush.

File: packages/angular_devkit/core/src/workspace/workspace-schema.ts [hunk 4]
```diff
@@ -20,7 +20,8 @@ export interface WorkspaceSchema {
    */
   newProjectRoot?: string;
   /**
-   * The default project.
+   * @deprecated From 7.0.0
+   * The default project.*
```
Comment (alan-agius4) [inline packages/angular_devkit/core/src/workspace/workspace-schema.ts]: Nit: you have an extra `*` at the end

Comment (filipesilva) [reply packages/angular_devkit/core/src/workspace/workspace-schema.ts]: Thanks, fixed

**Comment by alexeagle** [comment]
Blocked because we will reopen a larger discussion of project layout for ng8

**Comment by filipesilva** [comment]
We need to actually design what non-project scoped commands should do. Currently default project almost never applies, and when it does it might not be the right thing anyway.

**Comment by SchnWalter** [comment]
From what I can tell from https://github.com/angular/angular-cli/issues/11111#issuecomment-470635587, this PR should be closed. Although there are still inconsistencies between various commands where the project name is sometimes passed using `--projects` and other times it's `--name`. I'll look into this in the upcoming weeks and if it's clear what the default flag should be, I'll create some PRs to streamline the options.

**Comment by dgp1130** [comment]
Going through old PRs. After some discussion, we still think we should deprecate `defaultProject`, but this PR is fairly old and not usable in its current state anyways. Filed https://github.com/angular/angular-cli/issues/20661 to follow up with a deprecation.

**Pull request closed without merging**